system_setup/client/controllers/identity.py

```diff
@@ -0,0 +1,50 @@
+# Copyright 2015-2021 Canonical, Ltd.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import logging
+
+from subiquity.client.controllers import IdentityController
+from subiquity.common.types import IdentityData
+from system_setup.ui.views import WSLIdentityView
+
+log = logging.getLogger('system_setup.client.controllers.identity')
+
+
+class WSLIdentityController(IdentityController):
+
+    async def make_ui(self):
+        data = await self.endpoint.GET()
+        return WSLIdentityView(self, data)
+
+    def run_answers(self):
+        if all(elem in self.answers for elem in
+               ['realname', 'username', 'password']):
+            identity = IdentityData(
+                realname=self.answers['realname'],
+                username=self.answers['username'],
+                crypted_password=self.answers['password'])
+            self.done(identity)
+
+    def done(self, identity_data):
+        log.debug(
+            "IdentityController.done next_screen user_spec=%s",
+            identity_data)
+        if self.opts.dry_run:
+            username = "dryrun_user"
+        else:
+            username = identity_data.username
+        with open('/var/run/ubuntu_wsl_oobe_assigned_account', 'w') as f:
```

Comment:
Please protect this operation from dry run.
It should be possible with dry run to do so without running in 'sudo'
One alternate possibility here is that you could write this to a temp file or similar.

Reply:
Also, generally speaking, we want the server to be doing things like writing values and the client just sends requests to the server.
This will be important with the plans for using ubuntu-desktop-installer as a frontend - that client does not require root permissions to operate as the desktop installer with baseline subiquity, and if this were a client responsibility we would have to have this code that writes out the magic file both the UI and TUI versions of the client.